lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,40 @@
+class LazyHash < Hash
+
+  def method_missing(method_name, *args, &block)
+    if self.keys.include? method_name.to_s
+      self[method_name.to_s] = self[method_name.to_s].call
+    else
+      self.values
+    end
+  end
+
+  def lazy?(key)
+    self[key.to_s].respond_to? :call
```

Comment:
Good catch! Thanks for using `#respond_to?` 😃 
But what's happened if hash would store a symbol instead of a string? Please, modify it to manage both.